scylla/src/cloud/config.rs

```diff
@@ -191,4 +223,259 @@ mod deserialize {
             serde_yaml::from_reader(yaml)
         }
     }
+
+    fn get_pem_data_from_string_or_load_from_file(
+        data_name: &str,    // how data in string is called
+        path_name: &str,    // how data in file is called
+        data: Option<&str>, // data in string
+        path: Option<&str>, // path to data in file
+    ) -> Result<Box<[u8]>, CloudConfigError> {
+        let pem = if let Some(data) = data {
+            base64::decode(data)?
+        } else if let Some(path) = path {
+            let mut buf = vec![];
+            File::open(path)
+                .and_then(|mut f| f.read_to_end(&mut buf))
+                .map_err(|_| {
+                    CloudConfigError::Validation(format!(
+                        "Cannot read file at given {} {}.",
+                        path_name, path,
+                    ))
+                })?;
+            buf
+        } else {
+            return Err(CloudConfigError::Validation(format!(
+                "Either {} or {} has to be provided for authInfo.",
+                data_name, path_name,
+            )));
+        };
+        Ok(pem.into_boxed_slice())
+    }
+
+    impl TryFrom<RawCloudConfig> for super::CloudConfig {
+        type Error = CloudConfigError;
+
+        fn try_from(config: RawCloudConfig) -> Result<Self, Self::Error> {
+            if let Some(ref api_version) = config.apiVersion {
+                if !SUPPORTED_API_VERSIONS
+                    .iter()
+                    .any(|supported| supported == api_version)
+                {
+                    warn!(
+                        "Unknown API version: {}. Please update your driver.",
+                        api_version
+                    );
```

Comment:
This was @avelanarius suggestion. IIRC, it was to encourage users to update their driver if the change between driver versions wasn't great enough to render the driver unusable, but still big enough to justify such an admonition.

Reply:
OK. Technically speaking, a new version could change semantics of the existing fields so that they still parse but are inrepreted wrongly. However, if this is unlikely then I guess it's fine.